src/darwin/Framework/CHIP/MTRDeviceController_XPC.mm

```diff
@@ -43,6 +43,21 @@ @implementation MTRDeviceController_XPC
 
 @synthesize uniqueIdentifier = _uniqueIdentifier;
 
+- (NSXPCInterface *)_interfaceForServerProtocol
+{
+    NSXPCInterface * interface = [NSXPCInterface interfaceWithProtocol:@protocol(MTRXPCServerProtocol)];
+
+    NSSet * allowedClasses = [NSSet setWithArray:@[
+        [NSString class], [NSNumber class], [NSData class], [NSArray class], [NSDictionary class], [NSError class], [MTRCommandPath class], [MTRAttributePath class]
```

Comment:
Need to define encodeWithCoder for these classes?

Reply:
MTRCommandPath and MTRAttributePath (and MTREventPath) already implement `NSSecureCoding` and the `initWithCoder:` / `encodeWithCoder:` methods.